bindings/java/src/org/sleuthkit/datamodel/ScoringManager.java

```diff
@@ -114,47 +118,52 @@ private Score getAggregateScore(long objId, CaseDbConnection connection) throws
+	 * @param objId              Object id of the object.
 	 * @param dataSourceObjectId Data source object id.
-	 * @param score  Score to be inserted/updated.
-	 * @param transaction Transaction to use for the update.
+	 * @param score              Score to be inserted/updated.
+	 * @param version            Current version of score to be updated.
+	 * @param transaction        Transaction to use for the update.
 	 *
-	 * @throws TskCoreException
+	 * @throws TskCoreException If there is an error in core TSK.
+	 * @throws SQLException     If there is an SQL error, for example, the
+	 *                          record with specified version was not found.
 	 */
-	private void setAggregateScore(long objId, long dataSourceObjectId, Score score, CaseDbTransaction transaction) throws TskCoreException {
+	private void setAggregateScore(long objId, long dataSourceObjectId, Score score, int version, CaseDbTransaction transaction) throws TskCoreException, SQLException {
 		CaseDbConnection connection = transaction.getConnection();
-		setAggregateScore(objId, dataSourceObjectId, score, connection);
+		setAggregateScore(objId, dataSourceObjectId, score, version, connection);
 	}
 
 	/**
 	 * Inserts or updates the score for the given object.
 	 *
-	 * @param objId Object id of the object.
+	 * @param objId              Object id of the object.
 	 * @param dataSourceObjectId Data source object id.
-	 * @param score  Score to be inserted/updated.
-	 * @param connection Connection to use for the update.
+	 * @param score              Score to be inserted/updated.
+	 * @param version            Current version of score to be updated.
+	 * @param connection         Connection to use for the update.
 	 *
-	 * @throws TskCoreException
+	 * @throws TskCoreException If there is an error in core TSK.
+	 * @throws SQLException     If there is an SQL error, for example, the
+	 *                          record with specified version was not found.
 	 */
-	private void setAggregateScore(long objId, long dataSourceObjectId, Score score, CaseDbConnection connection) throws TskCoreException {
+	private void setAggregateScore(long objId, long dataSourceObjectId, Score score, int version, CaseDbConnection connection) throws TskCoreException, SQLException {
 
+		// insert or update score
 		String query = String.format("INSERT INTO tsk_aggregate_score (obj_id, data_source_obj_id, significance , confidence) VALUES (%d, %d, %d, %d)"
-				+ " ON CONFLICT (obj_id) DO UPDATE SET significance = %d, confidence = %d",
-				objId, dataSourceObjectId, score.getSignificance().getId(), score.getConfidence().getId(), score.getSignificance().getId(), score.getConfidence().getId() );
+				+ " ON CONFLICT (obj_id) DO UPDATE SET version = %d, significance = %d, confidence = %d WHERE version = %d",
+				objId, dataSourceObjectId, score.getSignificance().getId(), score.getConfidence().getId(), 
+				version + 1, score.getSignificance().getId(), score.getConfidence().getId(), version );
```

Comment:
In postgresql I got this error:
org.postgresql.util.PSQLException: ERROR: column reference "version" is ambiguous